components/scream/src/share/grid/abstract_grid.cpp

```diff
@@ -160,7 +160,7 @@ AbstractGrid::get_lid_to_idx_map () const {
 void AbstractGrid::
 set_geometry_data (const std::string& name, const geo_view_type& data) {
   // Sanity checks
-  EKAT_REQUIRE_MSG (data.extent_int(0)==m_num_local_dofs,
+  EKAT_REQUIRE_MSG (data.extent_int(0)==m_num_local_dofs || data.extent_int(0)==m_num_vert_levs,
```

Comment:
I wonder if the base class should do any check at all, at this point.

Reply:
Yeah, I would vote no. I'm sure there could be scenarios in the future where even more layouts could be beneficial.